### DISCUSSION TITLE
Toggleable button for smooth rotation

### DISCUSSION THREAD

**Discussion by arda4888** [Techokami/SonicWorldsNext#175] [Ideas]
I think a lot of people using Sonic Worlds Next would want to be able to easily change the smooth rotation feature. Right now the only way to do this (at least to my knowledge) is to change the code itself by removing a comment. I decided to add an in game menu to easily change this.

I added a global variable called smooth_rotation to global.gd

```
# the variable for controlling smooth rotation in game
var smooth_rotation: bool
```
I also added a save file and a save load.

```
file.set_value("SmoothRotation", "Is", smooth_rotation)

if file.has_section_key("SmoothRotation", "Is"):
	if (file.get_value("SmoothRotation", "Is")): 
		smooth_rotation = true
	else: smooth_rotation = false
```

Then I added the smooth rotation option to the menu. I think a lot of people would want this feature.

![image](https://github.com/user-attachments/assets/b4943cf9-a5c9-4c69-99aa-eb9eadd355f8)

**Comment by Raptorq1** [comment]
Did you have issues when adding the setting

**Reply by RepeatedKibbles** [reply]
For me, yes. It seems doesn't work and not showing it on settings (sorry if I sounded mean).
